src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -553,6 +553,26 @@ where
         self.expected_syscall_ptr.offset += get_syscall_size_from_name(syscall_name);
         Ok(syscall_request)
     }
+
+    fn replace_class(
+        &mut self,
+        vm: &mut VirtualMachine,
+        syscall_ptr: Relocatable,
+    ) -> Result<(), SyscallHandlerError> {
+        let request =
+            match self._read_and_validate_syscall_request("replace_class", vm, syscall_ptr) {
+                Ok(SyscallRequest::ReplaceClass(replace_class_request)) => replace_class_request,
+                _ => return Err(SyscallHandlerError::InvalidSyscallReadRequest),
+            };
+
+        let address = self.contract_address.clone();
+        self.starknet_storage_state
```

Comment:
replace class should not deploy the contract it should call to state.set_class_hash_at instead. Like it is done in cairo-lang

Reply:
Changed it to use `set_class_hash_at`